checker_tool/suites/eus_installer/general.sh

```diff
@@ -1,23 +1,198 @@
     fi
 }
 
-function success() {
-  msg "\33[32m[✔] ${1}\33[0m"
+# check if subscriptions in operator namespace are in correct version
+function check_subscriptions() { 
+    title "Checking subscriptions status in namespace: ${OPERATOR_NS}"
+    rc=0
+    oc get subscription.operators.coreos.com -n ${OPERATOR_NS}
+    oc get subscription.operators.coreos.com -n ${OPERATOR_NS} -o=jsonpath='{range .items[*]}{.metadata.name}{" "}{.status.state}{"\n"}{end}' | while read -r line; do
+        sub=$(echo $line | awk '{print $1}')
+        state=$(echo $line | awk '{print $2}')
+        if [[ "$state" != "AtLatestKnown" ]]; then
+            error "Subscription: ${sub} in namespace: ${OPERATOR_NS} not in Succeeded status"
+            append_check "eus_installer" "check subscription:${sub}" "failed" "Subscription: ${sub} in namespace: ${OPERATOR_NS} not in Succeeded status" ""
+            rc=1
+            # oc get subscription.operators.coreos.com ${sub} -n ${OPERATOR_NS} -oyaml | tee -a $logfile
+        fi
+    done
+
+    if [ $rc -eq 0 ]; then
+        success "All Subscriptions in namespace: ${OPERATOR_NS} in Succeeded status"
+        append_check "eus_installer" "check subscription" "ok" "All Subscriptions in namespace: ${OPERATOR_NS} in Succeeded status" ""
+    fi
+}
+
+# check if cert-manager, IAM, ODLM, CS-Operator subscriptions in ibm-common-services namespace are in correct version
+function check_sub_version(){
+    title "Checking subscriptions version in namespace: ${OPERATOR_NS}"
+    cs_operator_channel=$(oc get subscription.operators.coreos.com ibm-common-service-operator -n ${OPERATOR_NS} --ignore-not-found -o yaml | yq ".spec.channel")
+    iam_operator_channel=$(oc get subscription.operators.coreos.com ibm-iam-operator -n ${OPERATOR_NS} --ignore-not-found -o yaml | yq ".spec.channel")
+    odlm_channel=$(oc get subscription.operators.coreos.com operand-deployment-lifecycle-manager-app -n ${OPERATOR_NS} --ignore-not-found -o yaml | yq ".spec.channel")
+    cert_manager_channel=$(oc get subscription.operators.coreos.com ibm-cert-manager-operator -n ${OPERATOR_NS} --ignore-not-found -o yaml | yq ".spec.channel")
+    
+    # if not found cs-operator sub in operator namespace, check openshift-operators
+    if [[ "$cs_operator_channel" == "" ]]; then
+        cs_operator_channel=$(oc get subscription.operators.coreos.com ibm-common-service-operator -n openshift-operators --ignore-not-found -o yaml | yq ".spec.channel")
+    fi
+    # if not found cert-manager sub in operator namespace, check cs-control namespace
+    if [[ "$cert_manager_channel" == "" ]]; then
+        cert_manager_channel=$(oc get subscription.operators.coreos.com ibm-cert-manager-operator -n cs-control --ignore-not-found -o yaml | yq ".spec.channel")
+    fi
+
+    if [[ "$iam_operator_channel" == "stable-v1" ]] && [[ "$odlm_channel" == "stable-v1" ]] && [[ "$cs_operator_channel" == "stable-v1" ]] && [[ "$cert_manager_channel" == "stable-v1" ]]; then
```

Comment:
This is EUS group checks, so stable-v1 channel makes sense. I don't think we should be checking for `v3` or `v3.23` channels. Those should be under a new group for LTSR checks